presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SetFlatteningOptimizer.java

```diff
@@ -85,6 +86,34 @@ public PlanNode visitUnion(UnionNode node, RewriteContext<Boolean> context)
         }
 
         @Override
+        public PlanNode visitIntersect(IntersectNode node, RewriteContext<Boolean> context)
```

Comment:
This looks exactly the same as visitUnion except that you check that the source is an Intersect instead of union.  Is there a way to combine these into a common SetOperation method so we don't have duplicated code?